### ISSUE TITLE
[Impeller] `decodeImageFromPixels float32` does not work in impeller

### ISSUE THREAD

**Issue by dnfield** [flutter/flutter#135702]
The fourth color is expected to be `0x00ffffff` (transparent white) but it's coming out as transparent black.

I'm not really clear if this is an acceptable difference or not. @gaaclarke may know.

Found via https://github.com/flutter/flutter/issues/135693

**Comment by gaaclarke** [comment]
Sometimes it matters and sometimes it doesn't.  It depends on the blending that's happening.  I have a similar bug here where they are getting treated different in playgrounds versus skiagold: https://github.com/flutter/flutter/issues/135511

**Comment by dnfield** [comment]
I understand this can make a difference when blending, but is one or both of these values acceptable from the decoding of an image perspective?

**Comment by gaaclarke** [comment]
No, they aren't both acceptable.  They should match whatever you are inputing.  It's hard to verify where the problem is though.  Probably the easiest thing would be to do a `decodeImageFromPixels` call then `Image.toByteData` and look at the results.  If they are wrong there isn't an easy way to know if the problem is in `decodeImageFromPixels` or `toByteData` though without stepping through with a debugger.

**Comment by dnfield** [comment]
@jonahwilliams says we have some weirdness around how `Image.toByteData` is implemented for Impeller, so could be related to that.

**Comment by ColdPaleLight** [comment]
@dnfield 
I ran the code on the Android device, and I found that the results of the skia backend and the impeller backend were both 0x00000000. I guess the result of 0x00ffffff during testing is because in the flutter_tester environment, the io manager does not have a resource context, so the SkImage is not uploaded as a texture.

**Comment by jonahwilliams** [comment]
We think this is fixed.
